lib/Serialization/DeserializeSIL.cpp

```diff
@@ -598,6 +624,15 @@ SILFunction *SILDeserializer::readSILFunction(DeclID FID,
   if (fn->empty() && errorIfEmptyBody)
     return nullptr;
 
+  // Check that there are no unresolved forward definitions of opened
+  // archetypes.
+  for (auto &KV : OpenedArchetypesTracker.getOpenedArchetypeDefs()) {
+    if (!KV.getSecond() || isa<GlobalAddrInst>(KV.getSecond())) {
+      llvm_unreachable(
+          "All forward definitions of opened archetypes should be resolved");
+    }
+  }
```

Comment:
What about:

if (std::none_of(OpenedArchetypesTracker.getOpenedArchetypeDefs()),
     [](KVTy &KV) -> Bool { return !KV.getSecond() || isa<GlobalAddrInst>(KV.getSecond())) {
    llvm_unreachable("All forward definitions of opened archetypes should be resolved");
}